### DISCUSSION TITLE
Set a fixed train/val/(test) split of the training dataset

### DISCUSSION THREAD

**Discussion by rufeili** [talmolab/sleap#2648] [Help!] [answered]
Hi!
I was doing remote training via `export training job package` in the GUI given the recommendation [here](https://docs.sleap.ai/latest/guides/running-sleap-remotely/?h=remote). 
I'm curious if the random seed in the `Model Configuration/Data` block in the SLEAP GUI (version 1.6.1) affect things beyond train/val split of the training dataset?
Besides, it seems that we could do train/val/test split via CLI. Is this something we can also do in the GUI? If not, could I just change some config parameters in the .yaml file to achieve it? Or more generally speaking, any suggestions on best practices to specify a set train/val/test split for remote training.

Thanks,
Rufei

**Comment by gitttt-1234** [comment, marked as answer]
Hi @rufeili !

The `seed` in the Model Configuration/Data block affects more than just the train/val split. Here's the complete list:

1. **Train/Val Split** - The random sampling of frames into training vs. validation sets (when you don't provide separate val labels)
2. **Data Shuffling** - The order training batches are presented each epoch
3. **Weight Initialization** - How model parameters are initialized before training
4. **Data Augmentation** - The random augmentations (rotation, brightness, etc.) applied to each image
5. **Dropout** - Random dropout behavior during training

So setting a fixed seed will make your entire training run reproducible, not just the split.

Currently, the GUI only supports train/val splits via the `validation_fraction` setting. There's no built-in test split option in the GUI. However, you can try the below for achieving fixed train/val/test splits for remote training:

```bash
# Create 80/10/10 train/val/test split with embedded images
sleap split your_project.slp -o splits/ --train 0.8 --val 0.1 --test 0.1 --seed 42 --embed user

# This creates:
# - splits/train.pkg.slp
# - splits/val.pkg.slp
# - splits/test.pkg.slp
```

Optional: The `--embed user` flag embeds user-labeled frames into the output files, making them fully portable for remote training without needing the original videos.

Then edit your exported YAML config to point to these files:

```yaml
data_config:
  train_labels_path: ["./splits/train.pkg.slp"]
  val_labels_path: ["./splits/val.pkg.slp"]
  test_file_path: "./splits/test.pkg.slp"
```

Run `sleap split --help` for all available options.

Let us know if you have any other questions!

Thanks,

Divya

**Reply by rufeili** [reply]
thank you so much! this works well!
